sunbeam-python/sunbeam/plugins/ca/plugin.py

```diff
@@ -328,9 +349,58 @@ def tls_group(self) -> None:
     def ca_group(self) -> None:
         """Manage CA."""
 
+    @click.command()
+    @click.option(
+        "--format",
+        type=click.Choice([FORMAT_TABLE, FORMAT_YAML]),
+        default=FORMAT_TABLE,
+        help="Output format",
+    )
+    def list_outstanding_csrs(self, format: str) -> None:
+        """List outstanding CSRs"""
+        app = CA_APP_NAME
+        model = OPENSTACK_MODEL
+        action_cmd = "get-outstanding-certificate-requests"
+        jhelper = JujuHelper(self.deployment.get_connected_controller())
+        try:
+            action_result = get_outstanding_certificate_requests(app, model, jhelper)
+        except LeaderNotFoundException as e:
+            LOG.debug(f"Unable to get {self.app} leader to print CSRs")
+            raise click.ClickException(str(e))
+        except ActionFailedException as e:
+            LOG.debug(f"Running action {action_cmd} failed")
+            raise click.ClickException(str(e))
+
+        LOG.debug(f"Result from action {action_cmd}: {action_result}")
+        if action_result.get("return-code", 0) > 1:
+            raise click.ClickException(
+                "Unable to get outstanding certificate requests from CA"
+            )
+
+        csrs = {}
+        certs_to_process = json.loads(action_result.get("result")) or {}
+        for record in certs_to_process:
+            unit_name = record.get("unit_name")
+            unit_csrs = record.get("unit_csrs")
+            if unit_name:
+                csrs[unit_name] = unit_csrs
+
+        if format == FORMAT_TABLE:
+            table = Table()
+            table.add_column("Unit name")
+            table.add_column("CSR")
+            for unit, csr in csrs.items():
+                certs = (key.get("certificate_signing_request") for key in csr)
```

Comment:
Are we sure `certificate_signing_request` cannot be none ?

Reply:
In my testing, i have seen it never returned empty CSRs. But given the csrs are coming from external entity i.e., manual-tls-certificates charm, it is better to catch so handled the None case